Assets/Scripts/Models/Room.cs

```diff
@@ -370,34 +424,34 @@ public void ReadXml(XmlReader reader)
         }
     }
 
-    protected static void ActualFloodFill(Tile tile, Room oldRoom, int sizeOfOldRoom)
+    protected static Room ActualFloodFill(Tile tile, Room oldRoom, int sizeOfOldRoom)
```

Comment:
I would move DoRoomFloodFill, ActualFloodFill and EqualiseGasByTile into the new RoomManager class, since all those deal with possible multiple rooms. Neither of those functions should be static. To call then you would do something like ```World.Current.RoomManager.DoRoomFloodFill```.